### PR TITLE
dcp 727 fix handling of type error causing listener to break

### PR DESCRIPTION
**Pull request by prabh-t**

### CODE REVIEW

File: data/archiver/listener.py [hunk 1]
```diff
@@ -43,10 +43,15 @@ def _data_archiver_message_handler(self, body: dict, msg: Message):
 
             result = Archiver(ingest_cli, AwsS3()).start(req)            
             self.logger.info(f'Archived data for submission uuid {req.sub_uuid}')
-            
+
+        except ValueError as e:
+            error_msg = f'Invalid data archiving request {body}: {str(e)}'
+            self.logger.debug(error_msg)
+            result = DataArchiverResult(req.sub_uuid, success=False, error=error_msg)
```
Comment (ke4) [inline data/archiver/listener.py]: When would we get this `ValueError` exception? Are you sure that the `req` variable is not contains a `None` value?

**Review by ke4 — CHANGES_REQUESTED** [review]
I am already working on a change regarding this.
Anyway, please check my comments.
Thanks

File: data/archiver/listener.py (hunk 1 above)
Comment (prabh-t) [reply data/archiver/listener.py]: last push should fix issue.

File: data/archiver/listener.py [hunk 2]
```diff
@@ -32,22 +32,32 @@ def get_consumers(self, _consumer: Type[Consumer], channel) -> List[Consumer]:
...
         try:
+            if isinstance(body, str):
+                body = json.loads(body)
+
             req = DataArchiverRequest.from_dict(body)
+            sub_uuid = req.sub_uuid
             self.logger.info(f'Received data archiving request for submission uuid {req.sub_uuid}')
 
             result = Archiver(ingest_cli, AwsS3()).start(req)            
             self.logger.info(f'Archived data for submission uuid {req.sub_uuid}')
```
Comment (ke4) [inline data/archiver/listener.py]:
```suggestion
            self.logger.info(f'Archived data for submission uuid {sub_uuid}')
```

File: data/archiver/listener.py [hunk 3]
```diff
@@ -32,22 +32,32 @@ def get_consumers(self, _consumer: Type[Consumer], channel) -> List[Consumer]:
...
+    def _data_archiver_message_handler(self, body, msg: Message):
         ingest_cli = Ingest()
+        sub_uuid = None
         try:
+            if isinstance(body, str):
+                body = json.loads(body)
+
             req = DataArchiverRequest.from_dict(body)
+            sub_uuid = req.sub_uuid
             self.logger.info(f'Received data archiving request for submission uuid {req.sub_uuid}')
```
Comment (ke4) [inline data/archiver/listener.py]:
```suggestion
            self.logger.info(f'Received data archiving request for submission uuid {sub_uuid}')
```

**Review by ke4 — APPROVED** [review]
I approved it, but could you please apply the 2 suggested changes? Thanks

**Pull request merged**